### PR TITLE
Enhancing Error events for GET and UPDATE call to dataverse

### PR DESCRIPTION
**Pull request by BidishaMS**

### CODE REVIEW

File: src/web/client/WebExtensionContext.ts [hunk 1]
```diff
@@ -387,8 +387,9 @@ class WebExtensionContext implements IWebExtensionContext {
                     new Date().getTime() - requestSentAtTime,
                     response?.statusText,
                     '',
+                    telemetryEventNames.WEB_EXTENSION_API_REQUEST_FAILURE,
+                    response?.status.toString(),
                     telemetryEventNames.WEB_EXTENSION_POPULATE_LANGUAGE_ID_TO_CODE_API_ERROR,
```
Comment (tyaginidhi) [inline src/web/client/WebExtensionContext.ts:392]: we should not be using the telemetry event name as method name.

File: src/web/client/dal/remoteFetchProvider.ts [hunk 2]
```diff
@@ -147,8 +147,9 @@ async function fetchFromDataverseAndCreateFiles(
                     new Date().getTime() - requestSentAtTime,
                     errorMsg,
                     '',
-                    telemetryEventNames.WEB_EXTENSION_FETCH_DATAVERSE_AND_CREATE_FILES_API_ERROR,
-                    (error as Response)?.status.toString()
+                    telemetryEventNames.WEB_EXTENSION_API_REQUEST_FAILURE,
+                    (error as Response)?.status.toString(),
+                    telemetryEventNames.WEB_EXTENSION_FETCH_DATAVERSE_AND_CREATE_FILES_API_ERROR
```
Comment (tyaginidhi) [inline src/web/client/dal/remoteFetchProvider.ts:152]: same here and in all the updates above

File: src/web/client/telemetry/webExtensionTelemetry.ts [hunk 3]
```diff
@@ -86,7 +86,9 @@ export class WebExtensionTelemetry {
         duration?: number,
         errorMessage?: string,
         eventName?: string,
-        status?: string) {
+        status?: string,
+        methodName?: string) {
```
Comment (tyaginidhi) [inline src/web/client/telemetry/webExtensionTelemetry.ts:90]: this function signature update is correct - but we should make sure we are passing an actual methodname here - function.name property in javascript

**Review by tyaginidhi — CHANGES_REQUESTED** [review]

File: src/web/client/WebExtensionContext.ts (hunk 1 above)
Comment (BidishaMS) [reply src/web/client/WebExtensionContext.ts]: fixed

**Review by tyaginidhi — APPROVED** [review]

**Pull request merged**